client/plots/hierCluster.js

```diff
@@ -183,6 +189,12 @@ class HierCluster extends Matrix {
 			if (!(sampleId in d.samples)) continue
 			Object.assign(s, d.samples[sampleId])
 		}
+
+		// combine this.hierClusterSamples.refs.byTermId into this.data.refs.byTermId
+		const t = this.hierClusterSamples.refs.byTermId
+		for (const tw of this.hcTermGroup.lst) {
```

Comment:
@congyu-lu I just realized this should be `for(const $id of t)`, since we only want to add the data from `t = this.hierClusterSamples.refs.byTermId`. This would also avoid creating empty objects for terms that are neither in `d.refs.byTermId` and `t`. I'll approve for now, but you can also fix in this branch and I can re-approve.

Reply:
Thanks! Updated it.